connector/connect/docs/adding-proto-messages.md

```diff
@@ -0,0 +1,41 @@
+# Required, Optional and default values
+
+Connect adopts proto3, which does not support `required` constraint anymore. 
+For non-message proto fields, there is also no `has_field_name` functions to easy tell
+if a filed is set or not-set. (Read [proto3 field rules](https://developers.google.com/protocol-buffers/docs/proto3#specifying_field_rules))
+
+
+### Required field
+
+Even proto3 does not offer `required` constraint, there are still some fields that
+are semantically required. For such case, we shall add comment `(Required)` on the
+field to indicate it as required. The expectation for client implementation 
+(or any submitted proto plan) is that such fields should be always set, and server will
+always assume such fields are set and use whatever values from the fields directly.
+It is the client side's fault to not offer meaningful value for `required` field and in that case,
+the behavior on the server side is decided by the default value of the field.
```

Comment:
```suggestion
When adding fields that have required semantics, developers are required to follow
the outlined process. Fields that are semantically required for the server to 
correctly process the incoming message must be documented with `(Required)`. For scalar
fields the server will not perform any additional input validation. For compound fields,
the server will perform minimal checks to avoid null pointer exceptions but will not
perform any semantic validation.
```